recipes/msgpack-cxx/all/conanfile.py

```diff
@@ -23,18 +32,14 @@ def _source_subfolder(self):
     def _build_subfolder(self):
         return "build_subfolder"
 
-    def requirements(self):
-        self.requires("boost/1.78.0")
+    def configure_options(self):
+        # No boost was added in 4.1.0
+        if tools.Version(self.version) < "4.1.0":
+            del self.options.use_boost
 
-    def validate(self):
-        if self.options["boost"].header_only == True:
-            raise ConanInvalidConfiguration("{} requires boost with header_only = False".format(self.name))
-
-        if self.options["boost"].without_chrono == True or \
-            self.options["boost"].without_context == True or \
-            self.options["boost"].without_system == True or \
-            self.options["boost"].without_timer == True:
-            raise ConanInvalidConfiguration("{} requires boost with chrono, context, system, timer enabled".format(self.name))
+    def requirements(self):
+        if self.options.get_safe("use_boost", True):
```

Comment:
```suggestion
        if self.options.get_safe("use_boost", False):
```
the second parameter is the default value, which means, for any version, it always will install boost.

Reply:
No, boost is unconditionally required if version < 4.1.0